crates/rust-analyzer/src/tracing/config.rs

```diff
@@ -51,31 +53,48 @@ where
         let ra_fmt_layer = tracing_subscriber::fmt::layer()
             .with_target(false)
             .with_ansi(false)
-            .with_writer(writer)
-            .with_filter(filter);
+            .with_writer(writer);
 
-        let mut chalk_layer = None;
-        if let Some(chalk_filter) = self.chalk_filter {
-            let level: LevelFilter =
-                chalk_filter.parse().with_context(|| "invalid chalk log filter")?;
+        let chalk_layer = match self.chalk_filter {
+            Some(chalk_filter) => {
+                let level: LevelFilter =
+                    chalk_filter.parse().with_context(|| "invalid chalk log filter")?;
 
-            let chalk_filter = Targets::new()
-                .with_target("chalk_solve", level)
-                .with_target("chalk_ir", level)
-                .with_target("chalk_recursive", level);
-            chalk_layer = Some(
+                let chalk_filter = Targets::new()
+                    .with_target("chalk_solve", level)
+                    .with_target("chalk_ir", level)
+                    .with_target("chalk_recursive", level);
                 HierarchicalLayer::default()
                     .with_indent_lines(true)
                     .with_ansi(false)
                     .with_indent_amount(2)
                     .with_writer(io::stderr)
-                    .with_filter(chalk_filter),
-            );
+                    .with_filter(chalk_filter)
+                    .boxed()
+            }
+            None => HierarchicalLayer::default().with_filter(LevelFilter::OFF).boxed(),
         };
 
-        let profiler_layer = self.profile_filter.map(|spec| hprof::layer(&spec));
+        let profiler_layer = match self.profile_filter {
+            Some(spec) => hprof::layer(&spec).boxed(),
+            None => hprof::layer("").with_filter(LevelFilter::OFF).boxed(),
+        };
+
+        let json_profiling_filter = filter_fn(|metadata| {
+            let name = metadata.name();
+            let matches_name =
+                name == "vfs_load" || name == "parallel_prime_caches" || name == "discover_command";
```

Comment:
Why are you limiting to those spans? It seems more general to allow any span with filtering set by the user (`RA_PROFILE`).

Reply:
I was mostly interested in measuring startup, but I'll add another env variable for (`RA_PROFILE_JSON`?) to make this customizable.

(I meant to open this as a draft)

Reply:
I think it is fine to reuse `RA_PROFILE` for that.

Reply:
I respectfully disagree: I intend for the JSON output to be consumed by an LSP proxy that'll be taking these timings and converting them into metrics, and I don't really want to deal with the regular `RA_PROFILE` hprof output on the consumption side of things. I think `RA_PROFILE_JSON` should be used by machines, whereas I think `RA_PROFILE` should be used by people. I generally expect these to be mutually exclusive, tbh.

Reply:
I'll mark this as resolved, as I've added `RA_PROFILE_JSON`.